build failed

I following next steps:

```

ubuntu@vm-kzhuykov:~/dev$ git clone https://github.com/ZoneMinder/ZoneMinder.git ZMT
Cloning into 'ZMT'...
remote: Counting objects: 62294, done.
remote: Compressing objects: 100% (44/44), done.
remote: Total 62294 (delta 17), reused 0 (delta 0), pack-reused 62249
Receiving objects: 100% (62294/62294), 50.13 MiB | 1.65 MiB/s, done.
Resolving deltas: 100% (45555/45555), done.
Checking connectivity... готово.

ubuntu@vm-kzhuykov:~/dev$ cd ZMT
ubuntu@vm-kzhuykov:~/dev/ZMT$ git submodule update --init --recursive
Submodule 'web/api/app/Plugin/Crud' (https://github.com/FriendsOfCake/crud.git) registered for path 'web/api/app/Plugin/Crud'
Cloning into 'web/api/app/Plugin/Crud'...
remote: Counting objects: 47112, done.
remote: Compressing objects: 100% (32/32), done.
remote: Total 47112 (delta 21), reused 0 (delta 0), pack-reused 47078
Receiving objects: 100% (47112/47112), 10.49 MiB | 4.72 MiB/s, done.
Resolving deltas: 100% (34601/34601), done.
Checking connectivity... done.
Submodule path 'web/api/app/Plugin/Crud': checked out 'c3976f1478c681b0bbc132ec3a3e82c3984eeed5'

ubuntu@vm-kzhuykov:~/dev/ZMT$ docker build --no-cache .
Sending build context to Docker daemon 33.52 MB
Step 1 : FROM ubuntu:trusty
 ---> 0ccb13bf1954
Step 2 : MAINTAINER Kyle Johnson <kjohnson@gnulnx.net>
 ---> Running in 55b531aa641c
 ---> b41ed7ef6d23
Removing intermediate container 55b531aa641c

...................................................................................................

Step 7 : RUN make && make install
 ---> Running in 352b393100ea
Scanning dependencies of target zm
[  2%] Building CXX object src/CMakeFiles/zm.dir/zm_box.cpp.o
[  4%] Building CXX object src/CMakeFiles/zm.dir/zm_buffer.cpp.o
[  6%] Building CXX object src/CMakeFiles/zm.dir/zm_camera.cpp.o
[  8%] Building CXX object src/CMakeFiles/zm.dir/zm_comms.cpp.o
[ 10%] Building CXX object src/CMakeFiles/zm.dir/zm_config.cpp.o
[ 12%] Building CXX object src/CMakeFiles/zm.dir/zm_coord.cpp.o
[ 14%] Building CXX object src/CMakeFiles/zm.dir/zm_curl_camera.cpp.o
[ 16%] Building CXX object src/CMakeFiles/zm.dir/zm.cpp.o
[ 18%] Building CXX object src/CMakeFiles/zm.dir/zm_db.cpp.o
[ 20%] Building CXX object src/CMakeFiles/zm.dir/zm_logger.cpp.o
[ 22%] Building CXX object src/CMakeFiles/zm.dir/zm_event.cpp.o
[ 24%] Building CXX object src/CMakeFiles/zm.dir/zm_exception.cpp.o
[ 26%] Building CXX object src/CMakeFiles/zm.dir/zm_file_camera.cpp.o
[ 28%] Building CXX object src/CMakeFiles/zm.dir/zm_ffmpeg_camera.cpp.o
/ZoneMinder/src/zm_ffmpeg_camera.cpp: In member function 'int FfmpegCamera::OpenFfmpeg()':
/ZoneMinder/src/zm_ffmpeg_camera.cpp:241:67: error: 'av_dict_parse_string' was not declared in this scope
   ret = av_dict_parse_string(&opts, Options().c_str(), "=", ",", 0);
                                                                   ^
make[2]: *** [src/CMakeFiles/zm.dir/zm_ffmpeg_camera.cpp.o] Error 1
make[1]: *** [src/CMakeFiles/zm.dir/all] Error 2
make: *** [all] Error 2
The command '/bin/sh -c make && make install' returned a non-zero code: 2
ubuntu@vm-kzhuykov:~/dev/ZMT$
```

anyone can test this?

--- a/src/zm_ffmpeg.cpp
+++ b/src/zm_ffmpeg.cpp
@@ -70,7 +70,7 @@ enum _AVPIXELFORMAT GetFFMPEGPixelFormat(unsigned int p_colours, unsigned p_subp
   return pf;
 }
 /* The following is copied directly from newer ffmpeg. */
-#if LIBAVUTIL_VERSION_CHECK(52, 6, 0, 6, 0)
+#if LIBAVUTIL_VERSION_CHECK(52, 7, 0, 17, 100)
 #else
 static int parse_key_value_pair(AVDictionary **pm, const char **buf,
                                 const char *key_val_sep, const char *pairs_sep,

--- a/src/zm_ffmpeg.cpp
+++ b/src/zm_ffmpeg.cpp
@@ -69,6 +69,55 @@ enum _AVPIXELFORMAT GetFFMPEGPixelFormat(unsigned int p_colours, unsigned p_subp
 
   return pf;
 }
+/* The following is copied directly from newer ffmpeg. */
+#if LIBAVUTIL_VERSION_CHECK(52, 6, 0, 6, 0)
+#else
+static int parse_key_value_pair(AVDictionary **pm, const char **buf,
+                                const char *key_val_sep, const char *pairs_sep,
+                                int flags)
+{
+    char *key = av_get_token(buf, key_val_sep);
+    char *val = NULL;
+    int ret;
+
+    if (key && *key && strspn(*buf, key_val_sep)) {
+        (*buf)++;
+        val = av_get_token(buf, pairs_sep);
+    }
+
+    if (key && *key && val && *val)
+        ret = av_dict_set(pm, key, val, flags);
+    else
+        ret = AVERROR(EINVAL);
+
+    av_freep(&key);
+    av_freep(&val);
+
+    return ret;
+}
+int av_dict_parse_string(AVDictionary **pm, const char *str,
+                            const char *key_val_sep, const char *pairs_sep,
+                            int flags)
+   {
+       int ret;
+   
+       if (!str)
+          return 0;
+   
+       /* ignore STRDUP flags */
+       flags &= ~(AV_DICT_DONT_STRDUP_KEY | AV_DICT_DONT_STRDUP_VAL);
+   
+       while (*str) {
+           if ((ret = parse_key_value_pair(pm, &str, key_val_sep, pairs_sep, flags)) < 0)
+              return ret;
+   
+           if (*str)
+               str++;
+       }
+   
+       return 0;
+  }
+#endif
 #endif // HAVE_LIBAVUTIL
 
 #if HAVE_LIBSWSCALE && HAVE_LIBAVUTIL
@@ -239,5 +288,6 @@ int SWScale::ConvertDefaults(const uint8_t* in_buffer, const size_t in_buffer_si
   return Convert(in_buffer,in_buffer_size,out_buffer,out_buffer_size,default_input_pf,default_output_pf,default_width,default_height);
 }
 #endif // HAVE_LIBSWSCALE && HAVE_LIBAVUTIL
+
 
 #endif // HAVE_LIBAVCODEC || HAVE_LIBAVUTIL || HAVE_LIBSWSCALE

--- a/src/zm_ffmpeg.h
+++ b/src/zm_ffmpeg.h
@@ -32,6 +32,7 @@ extern "C" {
 #include <libavutil/avutil.h>
 #include <libavutil/base64.h>
 #include <libavutil/mathematics.h>
+#include <libavutil/avstring.h>
 
 /* LIBAVUTIL_VERSION_CHECK checks for the right version of libav and FFmpeg
  * The original source is vlc (in modules/codec/avcodec/avcommon_compat.h)
@@ -274,7 +275,15 @@ protected:
   #undef av_err2str
   #define av_err2str(errnum) av_make_error_string(errnum).c_str()
 
-  #endif // __cplusplus
+  /* The following is copied directly from newer ffmpeg */
+  #if LIBAVUTIL_VERSION_CHECK(52, 6, 0, 0, 0)
+  #else
+    int av_dict_parse_string(AVDictionary **pm, const char *str,
+                            const char *key_val_sep, const char *pairs_sep,
+                            int flags);
+  #endif
+
+#endif // __cplusplus
 
 
 #endif // ( HAVE_LIBAVUTIL_AVUTIL_H || HAVE_LIBAVCODEC_AVCODEC_H || HAVE_LIBAVFORMAT_AVFORMAT_H || HAVE_LIBAVDEVICE_AVDEVICE_H )

--- a/src/zm_ffmpeg.h
+++ b/src/zm_ffmpeg.h
@@ -276,7 +276,7 @@ protected:
   #define av_err2str(errnum) av_make_error_string(errnum).c_str()
 
   /* The following is copied directly from newer ffmpeg */
-  #if LIBAVUTIL_VERSION_CHECK(52, 6, 0, 0, 0)
+  #if LIBAVUTIL_VERSION_CHECK(52, 7, 0, 17, 100)
   #else
     int av_dict_parse_string(AVDictionary **pm, const char *str,
                             const char *key_val_sep, const char *pairs_sep,

